macros/sql/get_column_values.sql

```diff
@@ -36,16 +37,27 @@ Returns:
 
         {%- else -%}
 
-            select
-                {{ column }} as value
-
-            from {{ target_relation }}
-            group by 1
-            order by count(*) desc
-
-            {% if max_records is not none %}
-            limit {{ max_records }}
-            {% endif %}
+            with sorted_column_values as (
+
+                select
+                    {{ column }} as value,
+                    {%- if sort_column is none %}
+                    count(*) as sort_column
+                    {% else %}
+                    {# We take the max sort value for each value to make sure 
+                        there are no duplicate rows for each value #}
+                    max({{ sort_column }}) as sort_column
+                    {% endif %}
+                from {{ target_relation }}
+                group by 1
+                order by 2 {{ sort_direction if sort_direction else "desc" }}
+                {% endif %}
```

Comment:
fyi - I created a local copy of this and hit and error on DBT v18:

```
Encountered an error:
Compilation Error in macro (macros/sorted_get_column_values.sql)
  Got an unexpected control flow end tag, got endif but never saw a preceeding if (@ 63:8)
```

Reply:
Thanks for highlighting that! Just pushed a fix to remove that extra `endif`